zilliqa/src/consensus.rs

```diff
@@ -720,14 +720,18 @@ impl Consensus {
             } else {
                 let vote = self.vote_from_block(&block);
                 let next_leader = self.leader_at_block(&block, self.view.get_view());
-                self.create_next_block_on_timeout = false;
-                self.early_proposal = None;
 
                 let Some(next_leader) = next_leader else {
                     warn!("Next leader is currently not reachable, has it joined committee yet?");
                     return Ok(None);
                 };
 
+                if next_leader.public_key != self.public_key() {
+                    // Leader for this view is not us so clear any early work
+                    self.create_next_block_on_timeout = false;
+                    self.early_proposal = None;
```

Comment:
Is it possible that `early_proposal` contains some state which we do need to wipe even if we are the next proposer as well?

Reply:
It is possible, and is handled in `assemble_early_proposal_at()` by overwriting `early_proposal` unless:

```
        if self.early_proposal.is_some()
            && self.early_proposal.as_ref().unwrap().0.view() == self.view()
        {
            return Ok(());
        }
```